linera-views/src/indexed_db.rs

```diff
@@ -89,42 +104,45 @@ impl LocalReadableKeyValueStore<IndexedDbStoreError> for IndexedDbStore {
     }
 
     async fn read_value_bytes(&self, key: &[u8]) -> Result<Option<Vec<u8>>, IndexedDbStoreError> {
-        let key = js_sys::Uint8Array::from(key);
+        let key = self.extend_key(key);
+        let key = js_sys::Uint8Array::from(key.as_slice());
         let value = self.with_object_store(|o| o.get(&key))??.await?;
         Ok(value.map(|v| js_sys::Uint8Array::new(&v).to_vec()))
     }
 
     async fn contains_key(&self, key: &[u8]) -> Result<bool, IndexedDbStoreError> {
-        let key = js_sys::Uint8Array::from(key);
+        let key = self.extend_key(key);
+        let key = js_sys::Uint8Array::from(key.as_slice());
         let count = self.with_object_store(|o| o.count_with_key(&key))??.await?;
         assert!(count < 2);
         Ok(count == 1)
     }
 
     async fn contains_keys(&self, keys: Vec<Vec<u8>>) -> Result<Vec<bool>, IndexedDbStoreError> {
-        future::try_join_all(
-            keys.into_iter()
-                .map(|key| async move { self.contains_key(&key).await }),
-        )
+        future::try_join_all(keys.into_iter().map(|key| async move {
+            let key = self.extend_key(&key);
+            self.contains_key(&key).await
```

Comment:
Aren't you extending it twice now? Inside the `contains_key` call we're adding the root key again.

(Would be good to have a test that catches this.)

Reply:
You are right. Unfortunately, the indexed_db is not tested. Will see with James.